### PR TITLE
Add custom status support

### PR DESCRIPTION
**Pull request by PredaaA**
### Description of the changes

This PR adds support for custom status in `[p]set status` command group.
Custom status for bots are now supported since August 8th 2023 https://github.com/discord/discord-api-docs/pull/6345.

### Have the changes in this PR been tested?

Yes.

### CODE REVIEW

File: redbot/core/core_commands.py [hunk 1]
```diff
@@ -3212,6 +3212,43 @@ async def _set_my_status(self, ctx: commands.Context, status: discord.Status):
...
+        if activity:
+            await ctx.send(_("Custom status set to `{name}`.").format(name=name))
+        else:
+            await ctx.send(_("Custom status cleared."))
+
+    async def _set_my_status(self, ctx: commands.Context, status: discord.Status):
+        game = ctx.bot.guilds[0].me.activity if len(ctx.bot.guilds) > 0 else None
+        await ctx.bot.change_presence(status=status, activity=game)
+        return await ctx.send(_("Status changed to {}.").format(status))
+
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: This is unnecessary, it's already in our codebase looking exactly the same.
```suggestion
```

**Review by Jackenmen — CHANGES_REQUESTED** [review]
Can we make the argument name something like `text` instead of `name`? Also, the argument name in the Arguments section of the docstring is wrong.

**Review by Jackenmen — CHANGES_REQUESTED** [review]
You also need to update `docs/cog_guides/core.rst`.

**Comment by PredaaA** [comment]
> You also need to update `docs/cog_guides/core.rst`.

Done

File: docs/cog_guides/core.rst [hunk 2]
```diff
@@ -3769,6 +3769,35 @@ Maximum length for a competing status is 128 characters.
 **Arguments:**
     - ``[competing]`` - The text to follow ``Competing in``. Leave blank to clear the current activity status.
 
+.. _core-command-set-status-custom:
+
+""""""""""""""""""""
+set status custom
+""""""""""""""""""""
```
Comment (Jackenmen) [inline docs/cog_guides/core.rst]: nit, we keep the header's over/underline of the same length as the header text
```suggestion
"""""""""""""""""
set status custom
"""""""""""""""""
```

**Review by Jackenmen — APPROVED** [review]
LGTM.

**Comment by Rapptz** [comment]
I think this got merged too quick cause the API is still subject to change due to the awkwardness of it.

**Comment by Jackenmen** [comment]
I saw approval on the dapi docs PR so I assumed it's good to go 😬

**Comment by DJTOMATO** [comment]
Not sure if should be reported as a "bug"
but the latest red update allows us to set a custom status without a "activity as playing/listening, etc"

``!set status custom Invite me with !invitar | !help`` 

I have noticed that when music is being played by the bot, this command get resetted or cleared, having to set the status manually again.
https://bae.lena.moe/7wDTNa7VBjjo.png

**Pull request merged**